Virtraft changes

sjpotter opened the pull request:
these are changes fixing a couple of bugs in virtraft and making it better to debug with more controllable logging

bugs

1) _check_log_matching never worked in the presence of a failure (I actually think it still doesn't work, right, and should hard fail when it happens), but at least it doesn't a) silently fail when it fails and b) actually fail and ignore it.

now, it will print out when it fails and actually doesn't fail

2) entry_pop also just didn't work.  you aren't poping cfg change entries off the "targeted" node which is what the code thinks is happening, you are popping them (most likely) off a former leader that recv'd entries that are now invalid because leadership changed to another node.

sjpotter commented:
@yossigo this still needs to be applied, but has to be applied now after the work I've done to prepare for leader removal.  The idea is that we apply these fixes (rebased with the log type name change) and then we can try to remove the check that prevents leader removal and see what breaks (which now hopefully is nothing)

yossigo commented on tests/virtraft2.py:
```diff
@@ -1037,37 +1077,43 @@ def entry_pop(self, ety, ety_idx):
-        elif ety.type == lib.RAFT_LOGTYPE_ADD_NONVOTING_NODE:
-            if change.node_id == lib.raft_get_nodeid(self.raft):
+            if ety.type == lib.RAFT_LOGTYPE_DEMOTE_NODE:
+                server.set_connection_status(NODE_CONNECTED)
```
I guess this is just a temporary hack until we get rid of `RAFT_LOGTYPE_DEMOTE_NODE` completely, right?

sjpotter replied:
well, it is just a rename to of DEMOTE->REMOVE with removing the old REMOVE_NODE code (my other PR does that)

sjpotter commented on Makefile:
```diff
@@ -26,6 +26,7 @@ else
 SHAREDFLAGS = -shared
 SHAREDEXT = so
 endif
+LOG_LEVEL = $(or $(TEST),)
```
changed to your reccomendation

sjpotter commented on tests/virtraft2.py:
```diff
@@ -603,6 +624,8 @@ def diagnotistic_info(self):
             "Membership changes": self.num_membership_changes,
             "Compactions": self.num_compactions,
+            "Number of Active Servers": len(self.active_servers),
+            "active_server ids": [server.id for server in self.active_servers],
```
changed to your reccomendation

sjpotter commented:
@yossigo addressed all your comments but included one more bug fix.  virtraft now makes sure to set term to 1 (otherwise we could end up with snapshot term's of 0, which code fails on, perhaps code should be changed as well, but that requires a bit more thought, my gut feeling is that term should never able to be 0, the only Q is, should libraft force it to 1, or is that on caller to have to do correctly).
